med-inventory/src/components/purchase-medicine/purchase-layout/purchaseLayout.js

```diff
@@ -0,0 +1,135 @@
+        }, {
+          headerName: "PRICE", field: "mrp",sortable: true,  width:150, filter: true
+        },{
+          headerName : "CREATED BY", field :"CreatedBy",sortable: true, width:150, filter: true
+        },{
+          headerName : "CATEGORY" , field : "category" ,sortable : true , width:150, filter :true
+        },{
+          headerName : "CURRENT STOCK COUNT" , resizable :"true",field : "currentStockCount" , width:180, sortable : true , filter : true
+        },{
+          headerName : "NEW STOCK COUNT" , field : "newStockCount" , sortable : true , width:180, filter : true,
+          editable: true,cellEditor: "numericEditor",pinned: "right", lockPinned: true
+        }
+      ],
+      rowData: [],
+      rowSelection : "multiple",
+      singleEditClick : "true",
+      frameworkComponents: {
+        numericEditor: NumericEditor
+      },
+      selectedMedicineData : []
+    };
+  }
+  render(){
+      return(
+      <div>
+         {this.state.rowData  ? (
+            <Card className="card-style">
+                <CardContent>
+                 <div className="layout-style">
+                    <div
+                      className="ag-theme-balham grid-box"
+                      style={{
+                        height: "400px",
+                        width: "100%",
+                        padding: "50px"
+                      }}
+                    >
+                      <AgGridReact
+                              columnDefs={this.state.columnDefs}
+                              rowData={this.state.rowData}
+                              gridOptions={this.gridOptions}
+                              pagination="true"
+                              paginationPageSize="10"
+                              rowSelection={this.state.rowSelection}
+                              singleClickEdit={this.state.singleEditClick}
+                              frameworkComponents={this.state.frameworkComponents}
+                              onSelectionChanged={this.onSelectionChanged}
+                              onGridReady={this.onGridReady}
+                          >
+                      </AgGridReact>
+                    </div>
+                    <div className="preview-box">
+                      <PreviewPurchase medicinelistchoosen={this.state.selectedMedicineData} />
+                    </div>
+                  </div>
+                </CardContent>
+            </Card>
+         ):(
+           <CircularProgress />
+         )} 
+       </div>
+      );
+   }
+   onGridReady = (params) => {
+    this.gridApi = params.api;
+    this.gridColumnApi = params.columnApi;
+  };
+
+   componentWillMount() {
+    this.getMedicine();
+   } 
+
+   onSelectionChanged = (params) =>{
+    const selectedNodeList = params.api.getSelectedNodes();
+    let medicineList = [];
+    selectedNodeList.map((node) => {
+         medicineList.push({ ...node.data});
+         return console.log(node);
```

Comment:
Why are we returning console.log here?

Reply:
I am facing build issue, without any statement there...hence I added a console

Reply:
Replace the Map statement with this `selectedNodeList.map(node => medicineList.push({ ...node.data }));`
This should work fine

Reply:
Removed the console... Thanks for the suggestion